src/util-logopenfile.c

```diff
@@ -336,29 +361,253 @@ int SCConfLogReopen(LogFileCtx *log_ctx)
+ */
 static void SCLogFileCloseRedis(LogFileCtx *log_ctx)
 {
-    if (log_ctx->redis) {
-        redisReply *reply;
-        int i;
-        for (i = 0; i < log_ctx->redis_setup.batch_count; i++) {
-            redisGetReply(log_ctx->redis, (void **)&reply);
-            if (reply)
-                freeReplyObject(reply);
+    SCLogRedisContext * ctx = log_ctx->redis;
+
+    if ( ctx == NULL) {
+        return;
+    }
+
+    /* asynchronous */
+    if (log_ctx->redis_setup.async) {
+#if HAVE_LIBEVENT == 1
+        if (ctx->async != NULL) {
+            redisAsyncFree(ctx->async);
+        }
+        if (ctx->ev_base) {
+            event_base_loopbreak(ctx->ev_base);
+        }
+        ctx->async = NULL;
+#endif
+    } else {
+
+        /* synchronous */
+        if (ctx->sync) {
+            redisReply *reply;
+            int i;
+            for (i = 0; i < log_ctx->redis_setup.batch_count; i++) {
+                redisGetReply(ctx->sync, (void **)&reply);
+                if (reply)
+                    freeReplyObject(reply);
+            }
+            redisFree(ctx->sync);
+            ctx->sync = NULL;
+        }
+        log_ctx->redis_setup.tried = 0;
+        log_ctx->redis_setup.batch_count = 0;
+    }
+}
+
+static int SCConfLogReopenRedis(LogFileCtx *log_ctx);
+
+#if HAVE_LIBEVENT == 1
+
+#if HIREDIS_MAJOR == 0 && HIREDIS_MINOR < 11
+
+/** \brief RedisConnectCallback() Closes redis log more
+ *  \param c redis async context
+ */
+void RedisConnectCallback(const redisAsyncContext *c)
+{
+	SCLogInfo("Connected to redis server.");
+}
+#else
+
+/** \brief RedisConnectCallback() Closes redis log more
+ *  \param c redis async context
+ *  \param status status reported by async caller
+ */
+void RedisConnectCallback(const redisAsyncContext *c, int status)
+{
+	SCLogInfo("Connected to redis server. Status [%d]", status);
+}
+
+#endif // HIREDIS_MAJOR == 0 && HIREDIS_MINOR < 11
+
+/** \brief RedisDisconnectCallback() Callback when disconnection from redis happens.
+ *  \param c redis async context
+ *  \param status status reported by async caller
+ */
+void RedisDisconnectCallback(const redisAsyncContext *c, int status)
+{
+	SCLogInfo("Disconnected from redis server. Status [%d]", status);
+}
+
+
+/** \brief SCRedisAsyncCommandCallback() Callback when reply from redis happens.
+ *  \param c redis async context
+ *  \param r redis reply
+ *  \param privvata opaque datq with pointer to LogFileCtx
+ */
+static void SCRedisAsyncCommandCallback (redisAsyncContext *c, void *r, void *privdata)
+{
+
+    LogFileCtx *file_ctx = privdata;
+    SCLogRedisContext *ctx = file_ctx->redis;
+
+    redisReply *reply = r;
+
+    /* Disconnection or lost reply may have happened */
+    if (reply == NULL) {
+        if (ctx->ev_base != NULL) {
+            event_base_loopbreak(ctx->ev_base);
+        }
+        SCConfLogReopenRedis(file_ctx);
+        return;
+    }
+
+    if (reply->type == REDIS_REPLY_INTEGER ) {
```

Comment:
Please put test in debug preprocessor test to avoid something useless. In non debug mode SCLogDebug is just replace by an empty line.